provider/disk_utils.py

```diff
@@ -0,0 +1,159 @@
+"""
+Disk utilities function for managing disks and filesystems in guest environments.
+
+This module contains a function related to disk I/O test command execution on
+different OS types.
+"""
+
+from virttest import utils_disk
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def get_disk_by_serial(os_type, serial, session):
+    """
+    This function provides the disk ID reference.
+    It will vary based on the OS (ID or path).
+
+    :param os_type: The VM Operating System.
+    :param disk_id: The VM disk ID defined in the test cfg.
+    :param session: The guest session.
+    """
+    if os_type == "windows":
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+    else:
+        return get_linux_drive_path(session, serial)
+
+
+def init_disk_by_id(
+    params,
+    vm,
+    fstype=None,
+    os_type=None,
+    did="",
+):
+    """
+    Initializes the disk in the guest based on the received device ID.
+    By initialize, it means obtain the reference of the disk (ID or path) and
+    later in Linux systems, create the filesystem + mounting or formatting the
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    :param os_type: The VM Operating System.
+    :param did: The ID of the VM's disk.
+    """
+    session = vm.wait_for_login()
+    dst_dir = params.get("dst_dir")
+    image_params = params.object_params(did)
```

Comment:
It looks like there is confusion on the did and image name, did is disk id, it is the disk letter/index in OS, like sdb sdc in linux and 1,2,3 in windows, it is OS concept. The image name is defined in cfg, lt is logic name for the disk  like stg0 stg1. So for the function's parameter, it needs to pass the image name and did both from your code implementation. IMHO, just pass did that is fine due to we want to use whole disk ,  because  we do not care the image size. but in old code the image size is the input parameter, so the parameter include it is acceptable to me.  init_disk_by_id (params,     vm,    fstype=None,image,did) should meet your implementation.

Reply:
@qingwangrh so if we do not care about the image size how is it supposed to be obtained with the whole disk for configuring it? Just in case we can avoid more code...